rust/worker/src/blockstore/types.rs

```diff
@@ -297,6 +297,13 @@ impl<
         }
     }
 
+    pub(crate) async fn key_exists(&'referred_data self, prefix: &str, key: K) -> bool {
+        match self {
+            BlockfileReader::ArrowBlockfileReader(reader) => reader.key_exists(prefix, key).await,
+            BlockfileReader::MemoryBlockfileReader(reader) => todo!(),
```

Comment:
looks like we implemented this, does it need to be todo?

Reply:
good point! Missed updating here after implementing it